Adding rpv susy generator

kap01 opened the pull request:
I have made the required changes to be able to add the rpv susy generation to FairShip.
This includes modifying pythia8_conf.py, moving some common utility functions to both
HNL and SUSY generators into pythia8_conf_utils.py, introducing the script rpvsusy.py which
calculates widths and BFs as well as a set of .dat files which store the BFs for various neutralino mases. This is not the most flexible approach as any additional model would effectively have to
be also added into pythia8_conf as a separate configureXYZ function, 
making this script extremely long. 

BTW no need to pick up the changes to ShipAna.py not sure why that one looks so different to the master but please ignore this as i do not need to change anything to it

ThomasRuf commented on macro/run_simScript.py:
```diff
@@ -12,14 +12,16 @@
 eventDisplay = False
-inputFile    = "/eos/ship/data/Charm/Cascade-parp16-MSTP82-1-MSEL4-76Mpot_1.root"
+#inputFile    = "/eos/ship/data/Charm/Cascade-parp16-MSTP82-1-MSEL4-76Mpot_1.root"
+inputFile    = "/home/vagrant/ShipSoft/Cascade-parp16-MSTP82-1-MSEL4-76Mpot_1.root"
```
"/home/vagrant/ShipSoft/Cascade-parp16-MSTP82-1-MSEL4-76Mpot_1.root" looks like your private place.

ThomasRuf commented on macro/run_simScript.py:
```diff
@@ -12,14 +12,16 @@
 mcEngine     = "TGeant4"
 simEngine    = "Pythia8"  # "Genie" # Ntuple
-nEvents      = 100 
+nEvents      = 1000
```
would prefer to keep default 100 events, makes small test jobs faster.

ThomasRuf commented:
Hi Kostas,
I have asked Eric to have a look at the pythia8_conf changes, since he is the main author of it. For the other parts, unfortunately I am not able to figure out how to select specific files. This merge request for 6 commits looks suspicious to me:

invalid-email-address<https://github.com/invalid-email-address> and others added some commits 9 days ago <https://github.com/ShipSoft/FairShip/pull/16#commits-pushed-e0e9868>

[@invalid-email-address]<https://github.com/invalid-email-address>

invalid-email-address<https://github.com/invalid-email-address>

addition of rpv susy model and changes to pythia8_conf that reflect this<https://github.com/ShipSoft/FairShip/pull/16/commits/e0e98687b5bfa5c6a533f5ee620a31a993a69862>    what is invalid email address?

e0e9868<https://github.com/ShipSoft/FairShip/pull/16/commits/e0e98687b5bfa5c6a533f5ee620a31a993a69862>


kap01

added also temporary scripts for ShipAna and run_simScript<https://github.com/ShipSoft/FairShip/pull/16/commits/bfc333aab7e1c2a746c32bb02e93557f1de48ca0>     how to discard changes to ShipAna? Added some comments on Git about run_simScript. Not sure they are forwarded to you.

bfc333a<https://github.com/ShipSoft/FairShip/pull/16/commits/bfc333aab7e1c2a746c32bb02e93557f1de48ca0>


kap01

Merge branch 'master' of<https://github.com/ShipSoft/FairShip/pull/16/commits/a893923901e04603aa64c3509d0cc610814f57bf> https://github.com/ShipSoft/FairShip …    What does this mean? Don’t feel confident to merge this commit
Conflicts:
          macro/ShipAna.py

a893923<https://github.com/ShipSoft/FairShip/pull/16/commits/a893923901e04603aa64c3509d0cc610814f57bf>


kap01

commiting the dat files for susy rpv production<https://github.com/ShipSoft/FairShip/pull/16/commits/ff9794868cbff9bb7b245f4267848f62b5614495> this would be ok

ff97948<https://github.com/ShipSoft/FairShip/pull/16/commits/ff9794868cbff9bb7b245f4267848f62b5614495>


kap01

added options to run RPV susy model<https://github.com/ShipSoft/FairShip/pull/16/commits/e9a9c83e1a08b94a1c30e12f9917958a350c48b7> mmh, why are there + lines for ds==7, short muon shield?

e9a9c83<https://github.com/ShipSoft/FairShip/pull/16/commits/e9a9c83e1a08b94a1c30e12f9917958a350c48b7>


kap01

now pointing to eos file for charm root file<https://github.com/ShipSoft/FairShip/pull/16/commits/8bf3469ab0e98ae2f5a89186a1af7ba342dbbb9d> mmh, don’t understand at all



In summary, I would feel more confident if you could send me, email/afs, the modified files, rather than trusting GIT to make these merges.

Cheers,
thomas




Von: kap01 [mailto:notifications@github.com]
Gesendet: Donnerstag, 24. November 2016 08:54
An: ShipSoft/FairShip <FairShip@noreply.github.com>
Betreff: [ShipSoft/FairShip] Adding rpv susy generator (#16)


I have made the required changes to be able to add the rpv susy generation to FairShip.
This includes modifying pythia8_conf.py, moving some common utility functions to both
HNL and SUSY generators into pythia8_conf_utils.py, introducing the script rpvsusy.py which
calculates widths and BFs as well as a set of .dat files which store the BFs for various neutralino mases. This is not the most flexible approach as any additional model would effectively have to
be also added into pythia8_conf as a separate configureXYZ function,
making this script extremely long.

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/ShipSoft/FairShip/pull/16

Commit Summary

  *   addition of rpv susy model and changes to pythia8_conf that reflect this
  *   added also temporary scripts for ShipAna and run_simScript
  *   Merge branch 'master' of https://github.com/ShipSoft/FairShip
  *   commiting the dat files for susy rpv production
  *   added options to run RPV susy model
  *   now pointing to eos file for charm root file

File Changes

  *   M macro/ShipAna.py<https://github.com/ShipSoft/FairShip/pull/16/files#diff-0> (22)
  *   M macro/run_simScript.py<https://github.com/ShipSoft/FairShip/pull/16/files#diff-1> (55)
  *   M python/pythia8_conf.py<https://github.com/ShipSoft/FairShip/pull/16/files#diff-2> (239)
  *   A python/pythia8_conf_utils.py<https://github.com/ShipSoft/FairShip/pull/16/files#diff-3> (103)
  *   A python/rpvsusy.py<https://github.com/ShipSoft/FairShip/pull/16/files#diff-4> (439)
  *   A shipgen/branchingratiosrpvsusybench1.dat<https://github.com/ShipSoft/FairShip/pull/16/files#diff-5> (6402)
  *   A shipgen/branchingratiosrpvsusybench2.dat<https://github.com/ShipSoft/FairShip/pull/16/files#diff-6> (6402)
  *   A shipgen/branchingratiosrpvsusybench3.dat<https://github.com/ShipSoft/FairShip/pull/16/files#diff-7> (6402)
  *   A shipgen/branchingratiosrpvsusybench4.dat<https://github.com/ShipSoft/FairShip/pull/16/files#diff-8> (6402)
  *   A shipgen/branchingratiosrpvsusybench5.dat<https://github.com/ShipSoft/FairShip/pull/16/files#diff-9> (12804)

Patch Links:

  *   https://github.com/ShipSoft/FairShip/pull/16.patch
  *   https://github.com/ShipSoft/FairShip/pull/16.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/ShipSoft/FairShip/pull/16>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AGvbWU1KbF4DBGDB4hj2jYCpCXTy5i7Nks5rBUKhgaJpZM4K7Xay>.

kap01 commented:
Hey

Thanks for this. I swear I had gotten rid of parts from my private area in
run_simScript
I will send you the files. We might need to think a bit about how to change
run_simScript as I have
a feeling once multiple models start coming in, it will become quite an
unruly monolith. Ill give it some thought

Kostas

On 29 November 2016 at 08:33, Thomas Ruf <notifications@github.com> wrote:

> Hi Kostas,
> I have asked Eric to have a look at the pythia8_conf changes, since he is
> the main author of it. For the other parts, unfortunately I am not able to
> figure out how to select specific files. This merge request for 6 commits
> looks suspicious to me:
>
> invalid-email-address<https://github.com/invalid-email-address> and
> others added some commits 9 days ago <https://github.com/ShipSoft/
> FairShip/pull/16#commits-pushed-e0e9868>
>
> [@invalid-email-address]<https://github.com/invalid-email-address>
>
> invalid-email-address<https://github.com/invalid-email-address>
>
> addition of rpv susy model and changes to pythia8_conf that reflect this<
> https://github.com/ShipSoft/FairShip/pull/16/commits/
> e0e98687b5bfa5c6a533f5ee620a31a993a69862> what is invalid email address?
>
> e0e9868<https://github.com/ShipSoft/FairShip/pull/16/commits/
> e0e98687b5bfa5c6a533f5ee620a31a993a69862>
>
>
> kap01
>
> added also temporary scripts for ShipAna and run_simScript<https://github.
> com/ShipSoft/FairShip/pull/16/commits/bfc333aab7e1c2a746c32bb02e9355
> 7f1de48ca0> how to discard changes to ShipAna? Added some comments on Git
> about run_simScript. Not sure they are forwarded to you.
>
> bfc333a<https://github.com/ShipSoft/FairShip/pull/16/commits/
> bfc333aab7e1c2a746c32bb02e93557f1de48ca0>
>
>
> kap01
>
> Merge branch 'master' of<https://github.com/ShipSoft/FairShip/pull/16/
> commits/a893923901e04603aa64c3509d0cc610814f57bf>
> https://github.com/ShipSoft/FairShip … What does this mean? Don’t feel
> confident to merge this commit
> Conflicts:
> macro/ShipAna.py
>
> a893923<https://github.com/ShipSoft/FairShip/pull/16/commits/
> a893923901e04603aa64c3509d0cc610814f57bf>
>
>
> kap01
>
> commiting the dat files for susy rpv production<https://github.com/
> ShipSoft/FairShip/pull/16/commits/ff9794868cbff9bb7b245f4267848f62b5614495>
> this would be ok
>
> ff97948<https://github.com/ShipSoft/FairShip/pull/16/commits/
> ff9794868cbff9bb7b245f4267848f62b5614495>
>
>
> kap01
>
> added options to run RPV susy model<https://github.com/
> ShipSoft/FairShip/pull/16/commits/e9a9c83e1a08b94a1c30e12f9917958a350c48b7>
> mmh, why are there + lines for ds==7, short muon shield?
>
> e9a9c83<https://github.com/ShipSoft/FairShip/pull/16/commits/
> e9a9c83e1a08b94a1c30e12f9917958a350c48b7>
>
>
> kap01
>
> now pointing to eos file for charm root file<https://github.com/
> ShipSoft/FairShip/pull/16/commits/8bf3469ab0e98ae2f5a89186a1af7ba342dbbb9d>
> mmh, don’t understand at all
>
>
>
> In summary, I would feel more confident if you could send me, email/afs,
> the modified files, rather than trusting GIT to make these merges.
>
> Cheers,
> thomas
>
>
>
>
> Von: kap01 [mailto:notifications@github.com]
> Gesendet: Donnerstag, 24. November 2016 08:54
> An: ShipSoft/FairShip <FairShip@noreply.github.com>
> Betreff: [ShipSoft/FairShip] Adding rpv susy generator (#16)
>
>
> I have made the required changes to be able to add the rpv susy generation
> to FairShip.
> This includes modifying pythia8_conf.py, moving some common utility
> functions to both
> HNL and SUSY generators into pythia8_conf_utils.py, introducing the script
> rpvsusy.py which
> calculates widths and BFs as well as a set of .dat files which store the
> BFs for various neutralino mases. This is not the most flexible approach as
> any additional model would effectively have to
> be also added into pythia8_conf as a separate configureXYZ function,
> making this script extremely long.
>
> ________________________________
> You can view, comment on, or merge this pull request online at:
>
> https://github.com/ShipSoft/FairShip/pull/16
>
> Commit Summary
>
> * addition of rpv susy model and changes to pythia8_conf that reflect this
> * added also temporary scripts for ShipAna and run_simScript
> * Merge branch 'master' of https://github.com/ShipSoft/FairShip
> * commiting the dat files for susy rpv production
> * added options to run RPV susy model
> * now pointing to eos file for charm root file
>
> File Changes
>
> * M macro/ShipAna.py<https://github.com/ShipSoft/FairShip/
> pull/16/files#diff-0> (22)
> * M macro/run_simScript.py<https://github.com/ShipSoft/FairShip/
> pull/16/files#diff-1> (55)
> * M python/pythia8_conf.py<https://github.com/ShipSoft/FairShip/
> pull/16/files#diff-2> (239)
> * A python/pythia8_conf_utils.py<https://github.com/ShipSoft/
> FairShip/pull/16/files#diff-3> (103)
> * A python/rpvsusy.py<https://github.com/ShipSoft/FairShip/
> pull/16/files#diff-4> (439)
> * A shipgen/branchingratiosrpvsusybench1.dat<https://github.com/
> ShipSoft/FairShip/pull/16/files#diff-5> (6402)
> * A shipgen/branchingratiosrpvsusybench2.dat<https://github.com/
> ShipSoft/FairShip/pull/16/files#diff-6> (6402)
> * A shipgen/branchingratiosrpvsusybench3.dat<https://github.com/
> ShipSoft/FairShip/pull/16/files#diff-7> (6402)
> * A shipgen/branchingratiosrpvsusybench4.dat<https://github.com/
> ShipSoft/FairShip/pull/16/files#diff-8> (6402)
> * A shipgen/branchingratiosrpvsusybench5.dat<https://github.com/
> ShipSoft/FairShip/pull/16/files#diff-9> (12804)
>
> Patch Links:
>
> * https://github.com/ShipSoft/FairShip/pull/16.patch
> * https://github.com/ShipSoft/FairShip/pull/16.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub<https://github.com/
> ShipSoft/FairShip/pull/16>, or mute the thread<https://github.com/
> notifications/unsubscribe-auth/AGvbWU1KbF4DBGDB4hj2jYCpCXTy5i
> 7Nks5rBUKhgaJpZM4K7Xay>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ShipSoft/FairShip/pull/16#issuecomment-263507831>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AELFi5Fw_oivZ_3mAHGoChgeUksQ4C20ks5rC-NtgaJpZM4K7Xay>
> .
>

olantwin commented:
Hi Kostas, Thomas,

@kap01 I can guide you through cleaning up the commits in response to the comments using `git rebase` if you want. That can also be used to discard, say, the changes to ShipAna.py, while keeping the git information intact, so that others can later see that it's your code (which won't be possible if Thomas manually merges your changes).

I expect some of the spurious changes might be due to differing line endings. I've had some issues with that as well. Selecting individual lines of changed files using `git add -p` can help here.

Cheers,
Oliver
